### PR TITLE
Add basic support for ruff

### PR DESCRIPTION
**Pull request by ggravlingen**
I noticed `core` has started using this and I've done some testing in other repos during the week. It's really neat so I suggest we use it here too.

Adding basic support here and enabling a few rules.

### CODE REVIEW

**Comment by ggravlingen** [comment]
Having thought about it a bit, I guess we can remove `flake8` from the pre-commit rules and the CI/CD?

**Comment by MartinHjelmare** [comment]
Yes, I think we can remove flake8 and maybe isort too if we activate that in ruff?

File: pytradfri/__main__.py [hunk 1]
```diff
@@ -3,8 +3,8 @@
 
 import argparse
 import logging
-from pprint import pprint
 import uuid
+from pprint import pprint
```
Comment (MartinHjelmare) [inline pytradfri/__main__.py:7]: Can we add isort configuration to pyproject.toml to avoid changing the sorting?

Comment (MartinHjelmare) [reply pytradfri/__main__.py:7]: I think we may just need to make sure that ruff picks up the existing isort config?

Comment (ggravlingen) [reply pytradfri/__main__.py:7]: > Can we add isort configuration to pyproject.toml to avoid changing the sorting?

Yes, I'll add that.

Comment (ggravlingen) [reply pytradfri/__main__.py]: I thought ruff would extend config automatically. Maybe there's a flag for that?

Comment (ggravlingen) [reply pytradfri/__main__.py]: I did some digging in ruff's docs and GitHub issues section and couldn't find any obvious solutions but adding a separate section.

Comment (MartinHjelmare) [reply pytradfri/__main__.py]: Ok.

**Review by MartinHjelmare — APPROVED** [review]

**Pull request merged**